### ISSUE TITLE
serve: Angular module federation with  " buildLibsFromSource: false does not automatically build libs during serve " does not reflect changes into browser after first change

### ISSUE THREAD

**Issue by tomastrajan** [nrwl/nx#17921]
### Current Behavior

When;

1. serving `remote-a` with ` buildLibsFromSource: false` flag
2. making change to a library (lazy loaded feature) referenced in routing by `remote-a`
3. first change is rebuilt and refreshed in the browser
4. second change, library rebuilds, but the `remote-a` does not an d the change is not visible in the browser (even after manual refresh)

(similar autoclosed [issue](React application with buildLibsFromSource: false does not automatically build libs during serve #10124) in the past)

### Expected Behavior

All changes would behave the same

### GitHub Repo

https://github.com/angular-experts-io/module-federation-example

### Steps to Reproduce

1. `npm ci`
2. `npm run serve:remote-a`
3. make a change `libs/remote-a/feature/a/src/lib/remote-a-feature-a/remote-a-feature-a.component.ts`, will be reflected in the browser
4. another change to the same file, will not be reflected in the browser
5. workaround, remove "buildLibsFromSource": false, from project.json of the `remote-a`

Also described in the repo (bottom of the readme)

### Nx Report

```shell
Node   : 16.18.1
   OS     : linux-x64
   npm    : 8.19.2

   nx (global)        : 16.1.1
   nx                 : 16.4.2
   @nx/js             : 16.4.2
   @nx/jest           : 16.4.2
   @nx/linter         : 16.4.2
   @nx/workspace      : 16.4.2
   @nx/angular        : 16.4.2
   @nx/cypress        : 16.4.2
   @nx/devkit         : 16.4.2
   @nx/eslint-plugin  : 16.4.2
   @nrwl/tao          : 16.4.2
   @nx/web            : 16.4.2
   @nx/webpack        : 16.4.2
   typescript         : 5.1.6
   ---------------------------------------
   Local workspace plugins:
         @angular-experts/host-example/feature/dashboard-with-subroutes
         @angular-experts/host-example/feature/dashboard
         @angular-experts/remote-a/feature/a1
         @angular-experts/shared/ui/mf-loader
         @angular-experts/remote-a/feature/a
         @angular-experts/remote-b/feature/b
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by tomastrajan** [comment]
<img width="1355" alt="image" src="https://github.com/nrwl/nx/assets/3764868/dcd5b226-8279-4152-b6b0-bce9edbc45d2">
![image](https://github.com/nrwl/nx/assets/3764868/30487a56-87f2-4c83-8b80-1346ca441486)

**Comment by tomastrajan** [comment]
After checking with colleague, it seems to work as expected on his macbook, for me the ENV is Win 11 / WSL 2

**Comment by leosvelperez** [comment]
Thanks for reporting this! Apologies for the delay.

Is this still an issue today? Can you still reproduce this in newer versions of Nx?

**Comment by mirobo** [comment]
I've encountered the same issue also on Win 11 (but no WSL), had no time in investigating it seriously, so I just removed the option because it was becoming very slow: Once a changed code in a library it would re-check ALL libraries if they needed to be rebuilt, even if they had no "affection" with the changed library. And having a lot of libraries, the overhead is just too much.

When I want to do the production build of an app with `buildLibsFromSource=true`, NX will still run all other "build" targets from the buildable libraries because of the target defaults. But if I understand correctly they don't need to be built ever, if the libs should always be built from source. 

According to https://nx.dev/nx-api/angular/executors/dev-server#buildlibsfromsource the default is `buildLibsFromSource=true` but also having "build" depend o "^build" is the default.. I guess changing this could improve build times for a lot of people, wouldn't it? In the end each buildable library used `@nx/angular:ng-packagr-lite` which produces something which is not used by default?

So finally, I removed `"dependsOn": ["^build"],` in the `targetDefaults`  just to see if it still works.. and it seems that it does :D
